tests/logprob/test_order.py

```diff
@@ -0,0 +1,50 @@
+#   Copyright 2023 The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+#   MIT License
+#
+#   Copyright (c) 2021-2022 aesara-devs
+#
+#   Permission is hereby granted, free of charge, to any person obtaining a copy
+#   of this software and associated documentation files (the "Software"), to deal
+#   in the Software without restriction, including without limitation the rights
+#   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#   copies of the Software, and to permit persons to whom the Software is
+#   furnished to do so, subject to the following conditions:
+#
+#   The above copyright notice and this permission notice shall be included in all
+#   copies or substantial portions of the Software.
+#
+#   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+#   SOFTWARE.
+
+import pytensor.tensor as pt
+
+from pymc import logp
+from pymc.testing import assert_no_rvs
+
+
+def test_max():
+    x = pt.random.normal(0, 1, size=(3,))
+    x_name = "x"
+    x_max = pt.max_and_argmax(x, axis=-1)
+    x_max_value = pt.vector("x_max_value")
+    x_max_logprob = logp(x_max[0], x_max_value)
+
+    assert_no_rvs(x_max_logprob)
```

Comment:
We should test a couple of things more:
1. This fails when we ask for the logprob of argmax
2. This fails when an input is not a RandomVariable (just because it's a limitation for now)
3. This fails when the variable is not IID
4. This gives the expected result when evaluated

Reply:
These tests make a lot of sense. I have some questions regarding them. In case of argmax, the tests currently pass because ```pt.argmax``` creates a ```MaxAndArgmax``` op which is expected by the rewrite.

Reply:
That happens because the rewrite is not currently distinguishing which of the outputs is trying to be measured. You could have checked by seeing which of the two node outputs is actually present in `preserve_rv_mappings.needs_measuring`, and fail when it's the second (argmax). Regardless, the best I think is to apply that rewrite to transform `MaxAndArgmax` into simple `Max` or `Argmax` first and then this measurable rewrite only has to worry about `Max.`

We will probably remove `MaxAndArgmax` in the future in PyTensor anyway:  https://github.com/pymc-devs/pytensor/issues/334